### DISCUSSION TITLE
Virtual path

### DISCUSSION THREAD

**Discussion by scan-medics** [requarks/wiki#7082] [Help / Questions] [answered]
Hello everyone

The virtual path for a page has the restriction that the first level must not consist of two letters.
**ml**/new_page --> nok
**m1**/new_page --> ok
mlag/**dd**/new_page --> ok

However, the virtual path for an asset does not allow this further back.
mlag/**dd** --> nok
mlag/**d1** --> ok
mlag/**ddd** --> ok

I have understood that the restriction is due to the locals.

Questions 1:
The virtual path for the pages and assets is not checked immediately. Which version is correct?

For multilingual pages, the language is automatically added to the path, e.g.
en/mlag/dd/new_page
de/mlag/dd/new_page

Question 2:
What is best practice when storing assets if the same image is stored for each language?

**Comment by NGPixel** [comment, marked as answer]
There's no concept of locale for assets. Consider putting the locale as part of the filename, not the directory.

e.g. `foo/bar/some-image-en.png`, `foo/bar/some-image-fr.png`, etc.

**Reply by scan-medics** [reply]
Thanks for the answer.
